src/com/lilithsthrone/game/inventory/AbstractInventory.java

```diff
@@ -0,0 +1,109 @@
+package com.lilithsthrone.game.inventory;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * AbstractInventory serves to deduplicate code between the handling of Item, Clothing and Weapon in CharacterInventory.
+ *
+ * Package-privateness is a feature in that case.
+ * @param <T>
+ */
+class AbstractInventory<T extends AbstractCoreItem, U extends AbstractCoreType> {
+	private final Comparator<T> comparator;
+	private final Function<T, U> typeRetriever;
+	private Map<T, Integer> duplicateCounts;
+
+	AbstractInventory(Comparator<T> comparator, Function<T, U> typeRetriever) {
+		this.comparator = comparator;
+		this.typeRetriever = typeRetriever;
+		duplicateCounts = new LinkedHashMap<>();
+	}
+
+	public void clear() {
+		duplicateCounts.clear();
+	}
+
+	boolean isEmpty() {
+		return duplicateCounts.isEmpty();
+	}
+
+	public void sort() {
+		if (duplicateCounts.size() < 2) {
+			return;
+		}
+		List<T> weaponsToSort = new ArrayList<>(duplicateCounts.keySet());
```

Comment:
Minor gripe: if this is applicable to items, clothing, and weapons, then this variable shouldn't be called weaponsToSort.

Reply:
I'll get that changed on my end after merging.

Reply:
Fixed